### ISSUE TITLE
documentation decoding question

### ISSUE THREAD

**Issue by rschmaelzle** [nilearn/nilearn#1622]
Hi,
this may be a silly question - I am only embarking on using nilearn to perform some decoding: 
The [nilearn-documentation decoding intro](http://nilearn.github.io/decoding/decoding_intro.html) states (under section 2.1.2.2.1. Cross-validation) the common ML-mantra to keep train/test sets separate. The  code itself does not exactly contradict that, but I was confused by the following: In the [Haxby example](http://nilearn.github.io/auto_examples/02_decoding/plot_haxby_anova_svm.html#sphx-glr-auto-examples-02-decoding-plot-haxby-anova-svm-py) under section (8.3.7.5. Fit the decoder and predict), the anova_svm classifier is fit, and then in the next cell the x-val is performed. The result is an acc score of .97. I am irritated since to me this seems wrong. I may well be wrong - and would hope to learn why. Of note, a commented note states that the accuracy should be around .7, which I'd find way more credible. I played around with this code with my own data in which I did some buttonpresses to develop a trustworthy pipeline, and  there I am clearly overfitting it (100% acc)... My intuition would be that the classifier would have to be fit within each x-val loop separately on each fold of the dataset. Perhaps this is what's intended (or actually done elsewhere), but somehow I got confused. Any pointer to an example or explanations would be welcome. Thanks again for nilearn! 
R

**Comment by emdupre** [comment]
It looks like I accidentally changed the prediction categories from `['face', 'cat']` to `['face', 'house']` in 3f79b3e10c8808ba036a509982e7bfcaf2b0d271. Predicting for the correct category yields an accuracy of 0.7130, which is about what's seen in the commented code. I'll open a PR for this patch so it should be fixed soon-- sorry for the trouble !

**Comment by emdupre** [comment]
OK, I spoke too soon&mdash; `['face', 'house']` are the correct categories ! The error was that I had been subsetting `session` incorrectly, I think because I was encountering an odd error with `LeaveOneLabelOut` that I just now understand.

Apparently, `np.unique` (which is used in `sklearn.cross_validation.LeaveOneLabelOut`) has a known behavior of [flattening tuples](https://github.com/numpy/numpy/issues/4785). This wasn't an issue previously when using `np.recfromcsv` as the dtype is correctly set s.t. `np.unique` won't flatten the array. I've opened a PR in #1623 to fix this. Running with the implemented changes, I get the reported accuracy of 0.7037.

**Comment by KamalakerDadi** [comment]
Closing with #1623
